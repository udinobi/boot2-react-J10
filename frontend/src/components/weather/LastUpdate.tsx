import React from "react"

import { Option } from "ts-option"

import { frmt } from "./WeatherContainer"

import { MediumText } from "../lib/StyledText"

export interface Data {
    lastUpdate: Option<Date>
}

const days = [ "Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday" ]

const time = (dt: Date) => {
    const gmt = -Math.round(dt.getTimezoneOffset() / 60)
    const est = gmt < 0 ? "" : "+"
    return `${dt.getHours()}:${frmt(dt.getMinutes())}:${frmt(dt.getSeconds())} GMT${est}${gmt}`
}

export default (props: Data) => 
    props.lastUpdate
        .map(_ => (
            <MediumText>
                {days[_.getDay()]} {"\u00A0"}
                <span style={{ fontFamily: "Roboto, sans-serif" }}>
                    {time(_)}
                </span>
                {"\u00A0"} (last update)
            </MediumText>
        ))
        .getOrElse(<span />)
    
