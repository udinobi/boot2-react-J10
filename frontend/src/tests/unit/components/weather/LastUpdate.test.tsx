import { shallow } from "enzyme"
import toJson from "enzyme-to-json"
import React from "react"

import { none } from "ts-option"

import LastUpdate, { Data } from "../../../../components/weather/LastUpdate"

describe("--> Snapshot for Components > weather > 'LastUpdate'", () => {
    let props: Data

    beforeEach(() => {
        props = {
            lastUpdate: none
        }
    })

    it("matches", () => {
        const wrapper = shallow(<LastUpdate {...props}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
