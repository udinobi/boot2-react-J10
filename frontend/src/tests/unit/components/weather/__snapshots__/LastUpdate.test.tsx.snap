// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`--> Snapshot for Components > weather > 'LastUpdate' matches 1`] = `<span />`;
